components/match2/commons/opponent_display.lua

```diff
@@ -123,20 +125,18 @@ function OpponentDisplay.InlineOpponent(props)
 			template = opponent.template or 'tbd',
 		})
 
-	elseif opponent.type == 'literal' then
-		return opponent.name or ''
+	elseif Opponent.typeIsParty(opponent.type) then
+		return OpponentDisplay.InlineParty(props)
 
-	elseif opponent.type == 'solo' then
-		return OpponentDisplay.PlayerInlineOpponent(props)
+	else -- opponent.type == 'literal'
```

Comment:
If this `else` is always type `literal`, then make it an explicit `elseif`

Reply:
Are you looking to throw if the opponent type is not recognized? or do something else? 

Right now it returns the empty string on an unrecognized opponent type, which I think is reasonable. I can change it to throw if you want. Or return `nil`, which is similar to throwing because we almost never perform null checks on the result of display components.

Reply:
Unrecognized opponent type would indicate some pretty significant issues in the code and should be handled somehow.
Throwing an error message, like the code did before this PR, would at least allow the issue to be spotted, instead of hidden with a  `or ''`

Reply:
`OpponentDisplay.InlineOpponent` expects that the input opponent is a valid opponent. The type check call at the beginning of the function ensures the opponent is valid, assuming that type checking is enabled. Currently, type checking is disabled by default for performance reasons - it can be enabled with the `feature_type_check` flag.

Does that sufficiently address your concerns? Or do you still prefer an explicit check not hidden behind a feature flag?